src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -281,6 +237,14 @@ private void WatchForFileSystemEventsThreadStart()
 
             for (long i = 0; i < numEvents.ToInt32(); i++)
             {
+                // We need to special case the root path for pattern matching, so cache the path this way in case it is the root
+                // and use it for pattern matching. We'll go back to the eventPaths[i] to get the actual path for notifications
+                string path = eventPaths[i];
+                if (path.Equals(_fullDirectory, StringComparison.OrdinalIgnoreCase))
+                {
+                    path += System.IO.Path.DirectorySeparatorChar.ToString() + ".";
```

Comment:
Consider caching this into a static readonly string, e.g.
```C#
private static readonly string CurrentDirectorySuffix = Path.DirectorySeparatorChar + ".";
...
path += CurrentDirectorySuffix;
```
As it's currently written, not only will string.Concat be called by the compiler to produce a new string, but each call will end up allocating a temporary string containing "/" for the separator.